crates/sui-core/src/consensus_validator.rs

```diff
@@ -108,16 +111,33 @@ impl SuiTxValidator {
                 | ConsensusTransactionKind::CapabilityNotificationV2(_)
                 | ConsensusTransactionKind::RandomnessStateUpdate(_, _) => {}
 
-                ConsensusTransactionKind::UserTransaction(_)
-                | ConsensusTransactionKind::UserTransactionV2(_) => {
-                    if !epoch_store.protocol_config().mysticeti_fastpath() {
+                ConsensusTransactionKind::UserTransaction(_) => {
+                    if epoch_store.protocol_config().address_aliases()
+                        || epoch_store.protocol_config().disable_preconsensus_locking()
+                    {
+                        return Err(SuiErrorKind::UnexpectedMessage(
```

Comment:
Let's also reject the `ConsensusTransactionKind::CertifiedTransaction(certificate)` branch with `UnexpectedMessage` error, when `disable_preconsensus_locking()` is true.